pkg/virtctl/convert/guess.go

```diff
@@ -0,0 +1,36 @@
+package convert
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"k8s.io/client-go/pkg/util/yaml"
+	"unicode"
+)
+
+func GuessStreamType(source io.Reader, size int) (io.Reader, Type) {
+	if r, _, isXML := guessXMLStream(source, size); isXML {
+		return r, XML
+	} else if r, _, isJSON := yaml.GuessJSONStream(r, size); isJSON {
+		return r, JSON
+	} else {
+		return r, YAML
```

Comment:
Not really. For JSON, YAML and XML, we can just guess in every case. For JSON "{" (definitely not xml or yaml", for XML "<" (definitely not yaml or json).

If both don't match, I take the only possible left solution which will then later on fail when it is parsed.